### PR TITLE
[rllib] [docs] Cleanup RLlib API and make docs consistent with upcoming blog post

### PR DESCRIPTION
**Pull request by ericl**

### CODE REVIEW

File: doc/source/rllib-optimizers.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Here are the steps for using a RLlib policy optimizer with an existing algorithm.
+
+1. Implement the `Policy evaluator interface <rllib-dev.html#policy-evaluators-and-optimizers>`__.
+
+    - Here is an example of porting a `PyTorch Rainbow implementation <https://github.com/ericl/Rainbow/blob/rllib-example/rainbow_evaluator.py>`__.
+
+    - Another example porting a `TensorFlow DQN implementation <https://github.com/ericl/baselines/blob/rllib-example/baselines/deepq/dqn_evaluator.py>`__.
+
+2. Pick a `Policy optimizer class <https://github.com/ray-project/ray/tree/master/python/ray/rllib/optimizers>`__. The `LocalSyncOptimizer <https://github.com/ray-project/ray/blob/master/python/ray/rllib/optimizers/local_sync.py>`__ is a reasonable choice for local testing. You can also implement your own. Policy optimizers can be constructed using their ``make`` method (e.g., ``LocalSyncOptimizer.make(evaluator_cls, evaluator_args, num_workers, conf)``), or you can construct them by passing in a list of evaluators instantiated as Ray actors.
```
Comment (richardliaw) [inline doc/source/rllib-optimizers.rst]: One thing that would provide clarity is `conf` -> `optimizer_config`.

File: python/ray/rllib/optimizers/apex_optimizer.py [hunk 2]
```diff
@@ -110,13 +127,22 @@ def step(self):
...
+    """Main event loop of the Ape-X optimizer.
+
+    This class coordinates the data transfers between the learner thread,
+    remote evaluators (Ape-X actors), and replay buffer actors.
+
+    This optimizer requires that policy evaluators return an additional
+    "td_error" array in the info return of compute_gradients(). This error
+    term will be used for sample prioritization."""
 
     def _init(
```
Comment (richardliaw) [inline python/ray/rllib/optimizers/apex_optimizer.py]: One thing that is a little confusing and not very apparent in the docs is where all these parameters are being passed in. Reading this code, one would have to do some digging to jump through the various abstractions (ie, `ApexAgent` -^ `DQNAgent` -> `ApexOptimizer` -v `PolicyOptimizer` -^ `ApexOptimizer` to realize the chain of method calls needed to do this. 

Providing some note in the documentation page, and also a small comment here would be good.

Comment (richardliaw) [reply python/ray/rllib/optimizers/apex_optimizer.py]: After all, this is essentially exposed to user.

File: doc/source/rllib-optimizers.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - Option 1: call ``optimizer.step()`` from some existing training code. Training statistics can be retrieved by querying the ``optimizer.local_evaluator`` evaluator instance, or mapping over the remote evaluators (e.g., ``ray.get([ev.some_fn.remote() for ev in optimizer.remote_evaluators])``) if you are running with multiple workers.
+
+    - Option 2: define a full RLlib `Agent class <https://github.com/ray-project/ray/blob/master/python/ray/rllib/agent.py>`__. This might be preferable if you don't have an existing training harness or want to use features provided by `Ray Tune <tune.html>`__.
+
+
+Policy Optimizers
+-----------------
+
++-----------------------------+---------------------+-----------------+------------------------------+
+| **Policy optimizer class**  | **Operating range** | **Works with**  | **Description**              |
```
Comment (richardliaw) [inline doc/source/rllib-optimizers.rst]: I just built the docs locally, and this table is quite hard to read, especially with the need to horizontally scroll. Maybe just use sections, then add hyperlinks to relevant examples that actually use each optimizer..

File: doc/source/rllib-optimizers.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
+Using Policy Optimizers outside RLlib
```
Comment (richardliaw) [inline doc/source/rllib-optimizers.rst]: consider just renaming to Policy Optimizers

File: doc/source/rllib-optimizers.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+=====================================
+
+RLlib supports using its distributed policy optimizer implementations from external algorithms.
+
+Here are the steps for using a RLlib policy optimizer with an existing algorithm.
+
+1. Implement the `Policy evaluator interface <rllib-dev.html#policy-evaluators-and-optimizers>`__.
+
+    - Here is an example of porting a `PyTorch Rainbow implementation <https://github.com/ericl/Rainbow/blob/rllib-example/rainbow_evaluator.py>`__.
```
Comment (richardliaw) [inline doc/source/rllib-optimizers.rst]: explicit code examples here in this page would be good too

**Review by richardliaw — APPROVED** [review]
overall looks fine for a first pass

File: doc/source/rllib-optimizers.rst (hunk 4 above)
Comment (ericl) [reply doc/source/rllib-optimizers.rst]: Done

File: doc/source/rllib-optimizers.rst (hunk 1 above)
Comment (ericl) [reply doc/source/rllib-optimizers.rst]: Done

File: python/ray/rllib/utils/atari_wrappers.py [hunk 6]
```diff
@@ -144,11 +144,11 @@ def reset(self, **kwargs):
 
 
 class WarpFrame(gym.ObservationWrapper):
-    def __init__(self, env):
+    def __init__(self, env, dim):
         """Warp frames to 84x84 as done in the Nature paper and later work."""
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/utils/atari_wrappers.py]: Fixed

File: python/ray/rllib/utils/atari_wrappers.py [hunk 7]
```diff
@@ -185,7 +185,7 @@ def _get_ob(self):
         return np.concatenate(self.frames, axis=2)
 
 
-def wrap_deepmind(env, random_starts):
+def wrap_deepmind(env, random_starts=True, dim=80):
     """Configure environment for DeepMind-style Atari.
 
     Note that we assume reward clipping is done outside the wrapper.
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/utils/atari_wrappers.py]: Fixed

**Pull request merged**